e2e/lib.sh

```diff
@@ -81,35 +112,38 @@ block_until_ready() {
 
 # create_cluster launches a kind cluster and deploys Kilo, Adjacency, and a helper with curl.
 create_cluster() {
-	$KIND_BINARY delete clusters $KIND_CLUSTER > /dev/null
+	# shellcheck disable=SC2119
+	local CONFIG="${1:-$(build_kind_config)}"
+	_kind delete clusters $KIND_CLUSTER > /dev/null
 	# Create the kind cluster.
-	$KIND_BINARY create cluster --name $KIND_CLUSTER --config ./kind-config.yaml
+	_kind create cluster --name $KIND_CLUSTER --config <(echo "$CONFIG")
 	# Load the Kilo image into kind.
 	docker tag "$KILO_IMAGE" squat/kilo:test
+	# This command does not accept the --kubeconfig flag, so call the command directly.
 	$KIND_BINARY load docker-image squat/kilo:test --name $KIND_CLUSTER
 	# Create the kubeconfig secret.
-	$KUBECTL_BINARY create secret generic kubeconfig --from-file=kubeconfig="$KUBECONFIG" -n kube-system
+	_kubectl create secret generic kubeconfig --from-file=kubeconfig="$KUBECONFIG" -n kube-system
 	# Apply Kilo the the cluster.
-	$KUBECTL_BINARY apply -f ../manifests/crds.yaml
-	$KUBECTL_BINARY apply -f kilo-kind-userspace.yaml
+	_kubectl apply -f ../manifests/crds.yaml
+	_kubectl apply -f kilo-kind-userspace.yaml
 	block_until_ready_by_name kube-system kilo-userspace 
-	$KUBECTL_BINARY wait nodes --all --for=condition=Ready
+	_kubectl wait nodes --all --for=condition=Ready
 	# Wait for CoreDNS.
 	block_until_ready kube_system k8s-app=kube-dns
 	# Ensure the curl helper is not scheduled on a control-plane node.
-	$KUBECTL_BINARY apply -f helper-curl.yaml
+	_kubectl apply -f helper-curl.yaml
 	block_until_ready_by_name default curl
-	$KUBECTL_BINARY taint node $KIND_CLUSTER-control-plane node-role.kubernetes.io/master:NoSchedule-
-	$KUBECTL_BINARY apply -f https://raw.githubusercontent.com/heptoprint/adjacency/master/example.yaml
+	_kubectl taint node $KIND_CLUSTER-control-plane node-role.kubernetes.io/master:NoSchedule-
+	_kubectl apply -f https://raw.githubusercontent.com/heptoprint/adjacency/master/example.yaml
 	block_until_ready_by_name adjacency adjacency
 }
 
 delete_cluster () {
-	$KIND_BINARY delete clusters $KIND_CLUSTER
+	_kind delete clusters $KIND_CLUSTER --kubeconfig="$KUBECONFIG"
```

Comment:
Are we passing the `--kubeconfig` flag twice?

Reply:
totally. let me fix that